src/uu/expr/src/syntax_tree.rs

```diff
@@ -173,6 +175,87 @@ impl StringOp {
     }
 }
 
+/// Check errors with a supplied regular expression
+///
+/// GNU coreutils shows messages for invalid regular expressions
+/// differently from the oniguruma library used by the regex crate.
+/// This method attempts to do these checks manually in one pass
+/// through the regular expression.
+fn validate_regex(pattern: &str) -> ExprResult<()> {
+    let mut escaped_parens: u64 = 0;
+    let mut escaped_braces: u64 = 0;
+    let mut escaped = false;
+
+    let mut repeating_pattern_text = String::with_capacity(13);
+    let mut invalid_content_error = false;
+
+    for c in pattern.chars() {
+        match (escaped, c) {
+            (true, ')') => {
+                escaped_parens = escaped_parens
+                    .checked_sub(1)
+                    .ok_or(ExprError::UnmatchedClosingParenthesis)?;
+            }
+            (true, '(') => {
+                escaped_parens += 1;
+            }
+            (true, '}') => {
+                escaped_braces = escaped_braces
+                    .checked_sub(1)
+                    .ok_or(ExprError::UnmatchedClosingBrace)?;
+                let mut repetition = repeating_pattern_text[..repeating_pattern_text.len() - 1]
+                    .splitn(2, |x| x == ',');
+                match (repetition.next(), repetition.next()) {
+                    (None, None) => {
```

Comment:
i see 3 paths that the code coverage can't reach, is it possible to add tests to cover it? thanks